### ISSUE TITLE
JDK23 on Arm32: Build fails with Pattern.compile error thrown from JavacProcessingEnvironment.java

### ISSUE THREAD

**Issue by adamfarley** [adoptium/temurin-build#3586]
**What are you trying to do?**

Build JDK23 Temurin on Arm32.

**Expected behaviour:**

Build should be successful.

**Observed behaviour:**

https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk/job/jdk-linux-arm-temurin/274/console

```
13:57:01  Compiling up to 4 files for COMPILE_CREATE_SYMBOLS
13:57:01  Creating support/native/jdk.hotspot.agent/libsaproc/static/libsaproc.a from 10 file(s)
13:57:01  Creating support/modules_libs/jdk.hotspot.agent/libsaproc.so from 10 file(s)
13:57:02  Exception in thread "main" java.lang.ExceptionInInitializerError
13:57:02  	at jdk.compiler/com.sun.tools.javac.api.BasicJavacTask.initPlugins(BasicJavacTask.java:218)
13:57:02  	at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:294)
13:57:02  	at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:178)
13:57:02  	at jdk.compiler/com.sun.tools.javac.Main.compile(Main.java:66)
13:57:02  	at jdk.compiler/com.sun.tools.javac.Main.main(Main.java:52)
13:57:02  Caused by: java.util.regex.PatternSyntaxException: Unknown character property name {} near index 7
13:57:02  (\P{all})+
13:57:02         ^
13:57:02  	at java.base/java.util.regex.Pattern.error(Pattern.java:2204)
13:57:02  	at java.base/java.util.regex.Pattern.family(Pattern.java:3089)
13:57:02  	at java.base/java.util.regex.Pattern.sequence(Pattern.java:2332)
13:57:02  	at java.base/java.util.regex.Pattern.expr(Pattern.java:2245)
13:57:02  	at java.base/java.util.regex.Pattern.group0(Pattern.java:3229)
13:57:02  	at java.base/java.util.regex.Pattern.sequence(Pattern.java:2300)
13:57:02  	at java.base/java.util.regex.Pattern.expr(Pattern.java:2245)
13:57:02  	at java.base/java.util.regex.Pattern.compile(Pattern.java:1945)
13:57:02  	at java.base/java.util.regex.Pattern.<init>(Pattern.java:1576)
13:57:02  	at java.base/java.util.regex.Pattern.compile(Pattern.java:1101)
13:57:02  	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.<clinit>(JavacProcessingEnvironment.java:1750)
13:57:02  	... 5 more
13:57:02  Gendata.gmk:62: recipe for target '/home/jenkins/workspace/build-scripts/jobs/jdk/jdk-linux-arm-temurin/workspace/build/src/build/linux-arm-serverANDclient-release/buildtools/create_symbols_javac/_the.COMPILE_CREATE_SYMBOLS_batch' failed
```

**Any other comments:**

I looked into the code for JavacProcessingEnvironment.java line 1750, and it appears to be unchanged for 14 years. So l looked into the recent code commits for Pattern.java for JDK23, and it looked like the most likely cause (that hasn't been backported) was [this one](https://github.com/openjdk/jdk/commit/32efd23c5d59c03a6376c92c63f5947e961ee24e). 

I ran a [test build](https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk/job/jdk-linux-arm-temurin/275/console) with that [change removed](https://github.com/adamfarley/jdk_adopt/commit/75ff6ed33d6c8d93395dac22586abf74e1b6bc12), and the build passed. But then I reran my dev branch (0 difference between that and the code in the original failure), and [that build](https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk/job/jdk-linux-arm-temurin/276/) passed as well.

Based on this, I assume that the original failure was a one-off problem. Will [rerun the original build](https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk/job/jdk-linux-arm-temurin/277/) to verify that, and I'll close this bug right now. It has only been opened so that any future reproductions of the bug (if any) can match something in the git issue archives.

**Comment by adamfarley** [comment]
Ok, the original failed build failed again.

My current working theory is that the fix for this issue was merged after jdk-23+2, but it doesn't have a tag yet. Which is why adding a tag to my repository included all the changes that've gone into the upstream dev branch, which explains why my code worked with or without the fix, but builds against the Adoptium fork of the jdk source (with tag specified) didn't work.

To test this, I've rerun the Adoptium build without a specified tag level.

https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk/job/jdk-linux-arm-temurin/281/console

**Comment by adamfarley** [comment]
Yup, that worked.

TLDR: JDK-head on Arm32 shouldn't have this problem when we get JDK-23+3.
